scrapinghub/client/items.py

```diff
@@ -59,3 +73,30 @@ def _modify_iter_params(self, params):
         if offset:
             params['start'] = '{}/{}'.format(self.key, offset)
         return params
+
+    def list_iter(self, chunksize=10000, *args, **kwargs):
```

Comment:
I perfectly understand that the logic is a nice fit for iterating through huge items collections, but I'd suggest to have a lower default chunk size in the library, say 1k, for the general case. There're many parameters that should be taken into account when defining the specific value, like total amount of items and avg item size. Does it make sense?

Reply:
I'd say the average computer has still 8GB nowadays, and many budget notebooks have only 4GB. And it's not only about this: this is a thin client, it should be lightweight and fast, I wouldn't expect it eating even a few GBs of memory by default without my explicit approval. Also, having a pretty large chunk size means you need a relatively good network, which isn't often the case, so the decision regarding the chunk size should be conscious.

My key point here is that the default settings should fit an average user' demands and expectations, not an experienced one processing millions items in batches. It seems we have different opinions on "an average user", I'm OK with that. But as an experienced user you can use whatever you need by setting a custom chunk size value, I don't see a problem here.

Reply:
@manycoding @BurnzZ Guys, to keep it clear: I don't enforce decreasing the default size by any cost, it's just something I'd like to get a bit more attention and discuss rather than accepting as-is. From a practical point of view, a chunk of 100k items by 10KB gives approximately 1GB of memory, is it something you'd expect as users of the client library using default settings? Maybe I'm missing something here?

Reply:
Seems like a good time to send a questionnaire.

As an average user, I would use `.iter()` or work in an environment with enough memory first.

Reply:
Hey gents! To chime in the discussion here, the default `chunksize` of `10_000` was unconsciously used since it was working for our particular project, which has pretty much a typical number of items that one would see in project. Size per item is pretty much average I would say.

Now when choosing the default `chunksize` value, I think we'll need to set it to whatever value that **upholds** the objective of this brand new method, which is to prevent OOM issues faced by users when processing large jobs.

Speaking of large jobs, I have worked on projects which has **1 MB per item** which was insane. Consuming the jobs on those types of projects would really benefit in this new method we're creating. This makes me lean into choosing a lower `chunksize` altogether:

- `chunksize` of **1_000** * **1 MB** per item = **~1 GB**
- `chunksize` of **10_000** * **1 MB** per item = **~10 GB**

Lastly, I'm not sure if physical computers or laptops should be the basis for our benchmarks in setting this value. Nonetheless, we'll need to set the bar low. With that in mind, I'm proposing that we use **Scrapy Cloud** as the basis, where **1 Scrapy Cloud** unit would have **1 GB of memory**. Typical max SC units at the moment would be **6** which gives us a maximum of **6GB of memory** per job. (_P.S. We're using this heavily in Scrapy Cloud jobs so it would really make sense for us to make this as the basis_)

In the example above, using `chunksize=10_000` would result in **OOM** issues as **10 GB > 6 GB SC Maximum**. With that, I'm leaning towards setting `chunksize=1_000` as the default.

But before doing that, what do you guys think? @vshlapakov @manycoding